lib/components/app/view-switcher.tsx

```diff
@@ -0,0 +1,118 @@
+import { Button } from 'react-bootstrap'
+import { connect } from 'react-redux'
+import { FormattedMessage, injectIntl, IntlShape } from 'react-intl'
+import { withRouter } from 'react-router'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+
+import { MainPanelContent, setMainPanelContent } from '../../actions/ui'
+
+type Props = {
+  activePanel: number | null
+  intl: IntlShape
+  setMainPanelContent: (panel?: number | null) => void
+  sticky: boolean
+}
+
+/**
+ * This component is a switcher between
+ * the main views of the application.
+ */
+class ViewSwitcher extends Component<Props> {
+  static propTypes = {
+    activePanel: PropTypes.number,
+    setMainPanelContent: PropTypes.func,
+    sticky: PropTypes.bool
+  }
+
+  _showRouteViewer = () => {
+    this.props.setMainPanelContent(null)
+    this.props.setMainPanelContent(MainPanelContent.ROUTE_VIEWER)
+  }
+
+  _showTripPlanner = () => {
+    this.props.setMainPanelContent(null)
+  }
+
+  render() {
+    const { activePanel, intl, sticky } = this.props
+
+    const tripPlannerActive = activePanel === null
+    const routeViewerActive = activePanel === MainPanelContent.ROUTE_VIEWER
+
+    return (
+      <div
+        aria-label={intl.formatMessage({
+          id: 'components.ViewSwitcher.switcher'
+        })}
+        className="view-switcher"
+        id="view-switcher"
+        role="tablist"
```

Comment:
What else do you suggest? I think the tabs are better than buttons

Reply:
The other thing with tabs is that we would have to implement keyboard actions (arrows) too.

Reply:
This still needs to be resolved. I recommend just using `<div role="group">`. I don't think tabs apply in our situation because not one tab will be selected at all times. If you still want to use `tablist`, then implement the keyboard interactions per https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tablist_role#keyboard_interactions.